forem/forem
# Runtime Banner for Mobile Deep Linking

## What type of PR is this? (check all applicable)


## Description

Implements [RFC 72](https://github.com/forem/rfcs/pull/72)

## Related Tickets & Documents

https://github.com/forem/forem-ios/pull/47

## QA Instructions, Screenshots, Recordings

1. Boot app locally
2. Enable the runtime_banner Feature Flag
   - I like to use the console for this so Flipper.enable(:runtime_banner) would do the trick
3. Open any article, user page, homepage, etc on your **desktop browser**, **Android browser** or **Forem/DEV iOS app**
   - You shouldn't be able to see a banner with the text "Open with Forem app"
4. Open any page using your **iOS device**
   - You should be able to see a banner with the text "Open with Forem app"
5. Click on the banner and you should be able to see the deep linking flow

This is what it looks like right now **when the app is installed**:

https://user-images.githubusercontent.com/6045239/115251430-c8b31080-a0e7-11eb-9132-4161bd0bd4e7.MP4

This is what it looks like **when the app isn't installed**:

https://user-images.githubusercontent.com/6045239/113349663-a18dce00-92f5-11eb-8c98-5d07d2725308.MP4

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

Yes, I want the banner to be accessible but haven't made sure it is yet. Any suggestions to improve here are very welcome.

## Added tests?


## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: This feature will only go live the day we release version `1.0` of the Forem iOS app to the AppStore, that's the reason it's behind a `FeatureFlag`.

## [optional] Are there any post deployment tasks we need to perform?

None

## [optional] What gif best describes this PR or how it makes you feel?

This is me trying to make the banner to look "okay":

![css family guy](https://media.giphy.com/media/13FrpeVH09Zrb2/giphy.gif)
